dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -376,8 +388,9 @@ private long getBackoffNanoseconds() {
             if (failures == 0) {
                 return 0L;
             }
-            int upperBound = (int) Math.pow(2, failures - 1);
-            return Math.round(backoffSlotSize.toNanos() * jitter.getAsDouble() * upperBound);
+            int upperBound = (int) Math.pow(backoffFactor, failures - 1);
+            long backoffNanos = Math.round(backoffSlotSize.toNanos() * jitter.getAsDouble() * upperBound);
+            return Math.min(backoffNanos, maxBackoff.toNanos());
```

Comment:
This meaningfully complicates the backoff logic, making it harder to users to understand how the various configuration options affect behavior. I think it is important to keep the semantics of retry backoffs simple so it can be easily understood.

Why is not sufficient to modify the `backoffSlotSize` configuration?

Reply:
> just retry immediately on first retry of failure?

I don't think this is a good idea.
- For the vast majority of failures, you don't want this.
- There is not guarantee that a retried request will go to a different node. We probably want some amount of backoff time for the scores to updated so that it becomes less likely we send requests to the unavailable node.